supporter-product-data/src/main/scala/com/gu/services/S3Service.scala

```diff
@@ -24,32 +22,17 @@ object S3Service {
   def bucketName(stage: Stage) = s"supporter-product-data-export-${stage.value.toLowerCase}"
 
   def streamToS3(stage: Stage, filename: String, inputStream: InputStream, length: Long) = {
+    logger.info(s"Trying to stream to S3 - bucketName: ${bucketName(stage)}, filename: $filename, length: $length")
     val objectMetadata = new ObjectMetadata()
     objectMetadata.setContentLength(length)
+
     val putObjectRequest = new PutObjectRequest(bucketName(stage), filename, inputStream, objectMetadata)
-    val progressListener = new TransferProgressListener(filename)
-    putObjectRequest.setGeneralProgressListener(progressListener)
-    transferManager.upload(putObjectRequest)
-    progressListener.future
+    val transfer = transferManager.upload(putObjectRequest)
+
+    transfer.waitForCompletion()
```

Comment:
The changes in this file are because the previous implementation was swallowing the error so it was hard to diagnose what was going wrong